module #2. REST API Basics/rest-api-basic/core/src/main/java/com/epam/esm/repository/GiftCertificateRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.epam.esm.repository;
+
+import com.epam.esm.model.GiftCertificate;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface GiftCertificateRepository {
+    boolean isExists(GiftCertificate giftCertificate);
+    Long save(GiftCertificate giftCertificate);
+    Optional<GiftCertificate> findById(Long id);
+    Optional<GiftCertificate> findByName(String name);
+    Optional<List<GiftCertificate>> findAll();
+    void update(Long id, GiftCertificate giftCertificate);
+    void deleteById(Long id);
```

Comment:
All methods here is the same as in TagRepository with only one difference: the Object which will be returned. Please generify it.

Try to do the following: 

Create a new interface CrudRepository<...> which will take as generic arguments two parameters: type of Object on which repository will work on and type of its id (Integer, Long, String...). Move all methods from GiftCertificateRepository to CrudRepository<...>.

After that remove all methods in GiftCertificateRepository. Make GiftCertificateRepository implements CrudRepository<...> with required generic parameters.